lua/diagnostics.lua

```diff
@@ -36,7 +36,7 @@ module.sendAleResultsToDiagnostics = function(buffer, loclist)
     )
   end
 
-  local virtualtext_enabled_set = {['all'] = true, ['2'] = true, ['current'] = true, ['1'] = true}
+  local virtualtext_enabled_set = {['all'] = true, ['2'] = true, [2] = true, ['current'] = true, ['1'] = true, [1] = true}
```

Comment:
We shouldn't need this. I believe Lua turns numbers into strings for keys just like JS does.

Reply:
Hi w0rp, actually I opened this pr because here.

<img width="798" alt="image" src="https://user-images.githubusercontent.com/41264693/223737476-e46b612f-a2ee-43f3-8e66-1784314da8f1.png">
<img width="716" alt="image" src="https://user-images.githubusercontent.com/41264693/223737348-758b9d5b-7c6e-4728-af1f-21fad363c3f5.png">
<img width="729" alt="image" src="https://user-images.githubusercontent.com/41264693/223737899-bbeaf584-1289-45d0-90c3-7bb0eb2eafdc.png">

if I set g:ale_virtualtext_cursor as `1` instead of `'1'`, virtual text is not displayed.

you can simply reproduce the issue as follows

```vim
" with neovim
:let g:ale_use_neovim_diagnostics_api = 1
" this one show the virtual text
:let g:ale_virtualtext_cursor  = '1' | ALEDisableBuffer | ALEEnableBuffer
" thie one doesn't show the virtual text
:let g:ale_virtualtext_cursor  = 1 | ALEDisableBuffer | ALEEnableBuffer
```

Reply:
Aha! I was wrong! Lesson learned about Lua then. I'm not very familiar with Lua yet.